pennylane/ops/op_math/sum.py

```diff
@@ -458,3 +470,460 @@ def get_summands(self, cutoff=1.0e-12):
+
+OBS_MAP = {"PauliX": "X", "PauliY": "Y", "PauliZ": "Z", "Hadamard": "H", "Identity": "I"}
+
+
+def _compute_grouping_indices(observables, grouping_type="qwc", method="rlf"):
+    # todo: directly compute the
+    # indices, instead of extracting groups of observables first
+    observable_groups = qml.pauli.group_observables(
+        observables, coefficients=None, grouping_type=grouping_type, method=method
+    )
+
+    observables = copy(observables)
+
+    indices = []
+    available_indices = list(range(len(observables)))
+    for partition in observable_groups:  # pylint:disable=too-many-nested-blocks
+        indices_this_group = []
+        for pauli_word in partition:
+            # find index of this pauli word in remaining original observables,
+            for ind, observable in enumerate(observables):
+                if qml.pauli.are_identical_pauli_words(pauli_word, observable):
+                    indices_this_group.append(available_indices[ind])
+                    # delete this observable and its index, so it cannot be found again
+                    observables.pop(ind)
+                    available_indices.pop(ind)
+                    break
+        indices.append(tuple(indices_this_group))
+
+    return tuple(indices)
+
+
+class LinearCombination(Sum):
+    r"""TODO
+    """
+
+    num_wires = qml.operation.AnyWires
+    grad_method = "A"  # supports analytic gradients
+    batch_size = None
+    ndim_params = None  # could be (0,) * len(coeffs), but it is not needed. Define at class-level
+
+    def _flatten(self):
+        # note that we are unable to restore grouping type or method without creating new properties
+        return (self._coeffs, self._ops, self.data), (self.grouping_indices,)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        new_op = cls(data[0], data[1])
+        new_op._grouping_indices = metadata[0]  # pylint: disable=protected-access
+        new_op.data = data[2]
+        return new_op
+
+    def __init__(
+        self,
+        coeffs,
+        observables: List[Operator],
+        simplify=False,
+        grouping_type=None,
+        method="rlf",
+        _pauli_rep=None,
+        id=None,
+    ):
+        if qml.math.shape(coeffs)[0] != len(observables):
+            raise ValueError(
+                "Could not create valid LinearCombination; "
+                "number of coefficients and operators does not match."
+            )
+
+        self._coeffs = coeffs
+
+        self._ops = [convert_to_opmath(op) for op in observables]
+
+        self._hyperparameters = {"ops": self._ops}
+
+        self._grouping_indices = None
+
+        with qml.QueuingManager().stop_recording():
+            operands = [qml.s_prod(c, op) for c, op in zip(coeffs, observables)]
+
+        # TODO use grouping functionality of Sum once https://github.com/PennyLaneAI/pennylane/pull/5179 is merged
+        super().__init__(*operands, id=id, _pauli_rep=_pauli_rep)
+
+        if simplify:
```

Comment:
Simplifying upon initialization is a bit of an odd one. `Hamiltonian` originally supports it. This all feels a bit hacky, happy to hear feedback / suggestion on how to make this cleaner